### PR TITLE
Add external link check

### PR DESCRIPTION
**Pull request by yikeke**

### CODE REVIEW

File: .github/workflows/link-check.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      #    but if `-c` is too large then `Too Many Requests (HTTP error 429)` may be reported from `https://github.com/*`.
+      #  - we hardcode `--document-root` to `/github/workspace` in the container now.
+      #  - we use `http://172.*` as sample addresses in some docs, so we need to exclude them.
+      - name: Check external links
+        id: lc
+        uses: peter-evans/link-checker@v1
+        with:
+          args: -c 32 -d /github/workspace -r -x "http://172.*|https://github.com/.*/pull/.*|https://github.com/.*/issues/.*|https://github.com/[^.\n]*$|http://127.0.0.1.*|http://localhost.*|http://.*:.*|https://static.googleusercontent.com/media/research.google.com/.*|https://www.crunchbase.com/organization/.*|http://www.brendangregg.com.*|https://blog.csdn.net.*|http://mysql-cocoa.sourceforge.net.*|https://tools.ietf.org.*|http://code.google.com.*|https://spark.apache.org.*|https://www.caasdata.com/|https://www.linkedin.com/company/.*|https://sea-group.org/.*|https://octave.sourceforge.io/database/index.html|https://www.teld.cn/|http://gnade.sourceforge.net/|https://en.wikipedia.org/wiki/WeBank_\(China|https://twitter.com/PingCAP|.*.md" .
```
Comment (ran-huang) [inline .github/workflows/link-check.yaml:21]: Why are we excluding the following sites?

> https://www.crunchbase.com/organization/.*|http://www.brendangregg.com.*|https://blog.csdn.net.*|http://mysql-cocoa.sourceforge.net.*|https://tools.ietf.org.*|http://code.google.com.*|https://spark.apache.org.*|https://www.caasdata.com/|https://www.linkedin.com/company/.*|https://sea-group.org/.*|https://octave.sourceforge.io/database/index.html|https://www.teld.cn/|http://gnade.sourceforge.net/|

Comment (yikeke) [reply .github/workflows/link-check.yaml:21]: It's been tested in other repos that these links cause false alarms or links we want to ignore, so I keep them for this repo.

If you find some links that should not be excluded, I can remove them.

Comment (ran-huang) [reply .github/workflows/link-check.yaml:21]: Got it, thanks.

**Review by ran-huang — APPROVED** [review]

**Comment by CaitinChen** [comment]
@YiniXu9506 PTAL

**Comment by ran-huang** [comment]
Closing this pr because we have chosen another checker: https://github.com/pingcap/blog/pull/623

**Pull request closed without merging**